### PR TITLE
Restructure + nodenext compatibility

### PR DESCRIPTION
**Pull request by Uzlopak**
Part of [Migration of the infamous triplet to next level, Nodenext support ](https://github.com/fastify/fastify/issues/4349)

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by Fdawgs — APPROVED** [review]

**Comment by Uzlopak** [comment]
@Fdawgs 
Thank you

@Eomm 
@climba03003 

Could you please review the typescript changes?

File: index.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ function levelPlugin (fastify, opts, next) {
   fastify.level[name] = levelMore(path || name, opts.options, next)
 }
 
-module.exports = fp(levelPlugin, {
+module.exports = fp(fastifyLeveldb, {
   fastify: '4.x',
   name: '@fastify/leveldb'
 })
+module.exports.default = fastifyLeveldb
```
Comment (Eomm) [inline index.js:50]: Here the same doubt - I think we should really export the `fp()` one otherwise the function itself does not work with `register`

Comment (climba03003) [reply index.js:50]: Why? it is actually the same as follow.

```js

module.export = (function(plugin) {
  // it is direct mutation of plugin
  plugin[symbol] = metadata
  return plugin
})(fastifyLeveldb)

module.export.default = fastifyLeveldb // here is mutated function
```

Comment (Uzlopak) [reply index.js:50]: Thats imho wrong. Fastify-plugin mutates the function and sets the corresponding symbols. 

https://github.com/fastify/fastify-plugin/blob/b7c1190e72373879c7f14563dff67e6f7f03de96/plugin.js#L45

**Comment by Uzlopak** [comment]
@Eomm 
Can we release?

**Pull request merged**